kernel/kernel-api/src/main/java/io/delta/kernel/internal/ScanImpl.java

```diff
@@ -291,4 +299,11 @@ private Set<String> loadPartitionColNames() {
         }
         return Collections.unmodifiableSet(partitionColumnNames);
     }
+
+    private StructType loadDataSchema() {
```

Comment:
not really `loading` anything. `getDataSchema()` is better. but anyways, this probably belongs in the `Metadata` class, no? it already has `getSchema` and `getPartitionCols` ... we could easily add `getDataSchema` etc there too?

so it should just be `metadata.getDataSchema()`

Reply:
and this is what delta-spark does, too

Reply:
This relies on `loadPartitionColNames` which loads the partition columns from `ArrayValue`; I can move both over and make them lazy

Reply:
Yup! This is what delta-spark does. Much more cohesive.